packages/quantic/cypress/integration/facets/facet/facet.cypress.ts

```diff
@@ -0,0 +1,368 @@
+import {configure} from '../../../page-objects/configurator';
+
+import {FacetSelectors} from './facet-selectors';
+import * as Expectations from '../facet-common-expectations';
+import * as FacetExpectations from './facet-expectations';
+import {InterceptAliases, interceptSearch} from '../../../page-objects/search';
+import {checkFirstValue, checkLastValue} from './facet-actions';
+
+interface FacetOptions {
+  field: string;
+  label: string;
+  numberOfValues: number;
+  sortCriteria: string;
+  noSearch: boolean;
+}
+
+describe('Facet Test Suite', () => {
+  function visitFacetPage(options: Partial<FacetOptions> = {}) {
+    interceptSearch();
+
+    cy.visit(`${Cypress.env('examplesUrl')}/s/quantic-facet`);
+    configure(options);
+  }
+
+  describe('with values', () => {
+    const defaultField = 'objecttype';
+    const defaultLabel = 'Type';
+    const defaultNumberOfValues = 8;
+
+    function setupWithValues() {
+      visitFacetPage({
+        field: defaultField,
+        label: defaultLabel,
+        numberOfValues: defaultNumberOfValues,
+      });
+    }
+
+    describe('verify rendering', () => {
+      before(setupWithValues);
+
+      Expectations.expectLabelContains(FacetSelectors, defaultLabel);
```

Comment:
Given it appears we're passing FacetSelectors on every call to a predefined expectation, could there be a way to remove this? Perhaps executing a function in the setup that binds all expectations to an object with a defined `selectors` member set to `FacetSelectors`?
Or maybe this is a special case and we'll commonly want to use varying selectors in a given test suite. Curious to know your thoughts.

Reply:
I asked myself the same question doing it :smile: . I thought about using a builder function that would receive `FacetSelectors` and would return the expectations. The problem with this approach is that expectations are using different selector types (because exepectations depend on various facet capabilities). We could end up with the same type growing issue I mentioned in another comment.

However, exposing a builder function for each selector type (aka., facet capabilities), we could compose them in the test suite in a single expectation object.

I'll give it a try. Thanks for asking!

Reply:
I guess it would work. Here is the proposed solution:

    // facet-common-expectations.ts
    export function baseFacetExpectations(selector: BaseFacetSelector) {
        return {
            labelContains: (label: string) => {
                it(...)
            },
            // other base expectations goes here
        };
    }

    export function facetWithCheckboxExpectations(selector: FacetWithCheckboxSelector) {
        ...
    }

    //  facet-expectations.ts
    export const FacetExpectations = {
        ...baseFacetExpectations(FacetSelector),
        ...facetWithCheckboxExpectations(FacetSelector),
        ...
    };

    // facet.cypress.ts
    import {FacetExpectations as Expect}  from './facet-expectations';
    ...
    describe('verify rendering', () => {
        before(setupWithValues);

       Expect.labelContains(defaultLabel);
       ...
    }
    
It requires some more code to declare the expectations and merge them for the test suite, but the tests are quite clean.

I guess it is worth it. Thoughts?

Reply:
I think it's pretty nice!!